### PR TITLE
Global Object

### PR DESCRIPTION
**Pull request by AbmSourav**

### CODE REVIEW

**Comment by jamal-pb95** [comment]
Sorry vai, its been too late. But Congrats!!! you have solved your issue. Thanks for your contribution.

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 1]
```diff
@@ -1,87 +1,87 @@
 
-# Global object
+# গ্লোবাল অবজেক্ট 
 
-The global object provides variables and functions that are available anywhere. By default, those that are built into the language or the environment.
+গ্লোবাল অবজেক্টটি ভেরিয়েবলস্‌ এবং ফাংশনস্‌ প্রোভাইড করে যা, যে কোনও জায়গায় উপলব্ধ। ডিফল্টভাবে, সেগুলো ল্যাঙ্গুইজ বা ইনভার্মেন্টের মধ্যে তৈরি হয়।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:4]:
```suggestion
গ্লোবাল অবজেক্টটি ভেরিয়েবলস্‌ এবং ফাংশনস্‌ প্রোভাইড করে যা, যে কোনও জায়গায় উপলব্ধ। ডিফল্টভাবে, সেগুলো ল্যাঙ্গুইজ বা এনভাইরনমেন্টের মধ্যে তৈরি হয়।
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 2]
```diff
@@ -1,87 +1,87 @@
 
-# Global object
+# গ্লোবাল অবজেক্ট 
 
-The global object provides variables and functions that are available anywhere. By default, those that are built into the language or the environment.
+গ্লোবাল অবজেক্টটি ভেরিয়েবলস্‌ এবং ফাংশনস্‌ প্রোভাইড করে যা, যে কোনও জায়গায় উপলব্ধ। ডিফল্টভাবে, সেগুলো ল্যাঙ্গুইজ বা ইনভার্মেন্টের মধ্যে তৈরি হয়। 
 
-In a browser it is named `window`, for Node.js it is `global`, for other environments it may have another name.
+ব্রাউজারে এটিকে `window` বলা হয়, নোড জেএসে `global` বলা হয়, এছাড়া অন্যান্য ইনভার্মেন্টে পৃথক নাম থাকতে পারে।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:6]:
```suggestion
ব্রাউজারে এটিকে `window` বলা হয়, নোড জেএসে `global` বলা হয়, এছাড়া অন্যান্য এনভাইরনমেন্টে পৃথক নাম থাকতে পারে। 
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 3]
```diff
@@ -1,87 +1,87 @@
...
+# গ্লোবাল অবজেক্ট 
 
-The global object provides variables and functions that are available anywhere. By default, those that are built into the language or the environment.
+গ্লোবাল অবজেক্টটি ভেরিয়েবলস্‌ এবং ফাংশনস্‌ প্রোভাইড করে যা, যে কোনও জায়গায় উপলব্ধ। ডিফল্টভাবে, সেগুলো ল্যাঙ্গুইজ বা ইনভার্মেন্টের মধ্যে তৈরি হয়। 
 
-In a browser it is named `window`, for Node.js it is `global`, for other environments it may have another name.
+ব্রাউজারে এটিকে `window` বলা হয়, নোড জেএসে `global` বলা হয়, এছাড়া অন্যান্য ইনভার্মেন্টে পৃথক নাম থাকতে পারে। 
 
-Recently, `globalThis` was added to the language, as a standardized name for a global object, that should be supported across all environments. In some browsers, namely non-Chromium Edge, `globalThis` is not yet supported, but can be easily polyfilled.
+সম্প্রতি, গ্লোবাল অবজেক্টের আদর্শায়িত নামের জন্য ল্যাঙ্গুইজে `globalThis` এড করা হয়েছে, যা সমস্ত ইনভার্মেন্টে সাপোর্ট করার কথা। কিছু ব্রাউজার, যেমন নন-ক্রোমিয়াম এজ, এখনও `globalThis` সাপোর্ট করে না, তবে সহজে পলিফিল করা যাবে।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:8]:
```suggestion
সম্প্রতি, গ্লোবাল অবজেক্টের আদর্শায়িত নামের জন্য ল্যাঙ্গুইজে `globalThis` এড করা হয়েছে, যা সমস্ত এনভাইরনমেন্টে সাপোর্ট করার কথা। কিছু ব্রাউজার, যেমন নন-ক্রোমিয়াম এজ, এখনও `globalThis` সাপোর্ট করে না, তবে সহজে পলিফিল করা যাবে।
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 4]
```diff
@@ -1,87 +1,87 @@
...
 
-We'll use `window` here, assuming that our environment is a browser. If your script may run in other environments, it's better to use `globalThis` instead.
+আমরা এখানে `window` ব্যবহার করব, ধরে নিচ্ছি আমাদের ইনভার্মেন্ট হল ব্রাউজার। আপনার স্ক্রিপ্ট যদি অন্যান্য ইনভার্মেন্টে রান হয়, তাহলে `globalThis` ব্যবহার করাই ভাল।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:10]:
```suggestion
আমরা এখানে `window` ব্যবহার করব, ধরে নিচ্ছি আমাদের এনভাইরনমেন্ট হল ব্রাউজার। আপনার স্ক্রিপ্ট যদি অন্যান্য এনভাইরনমেন্টে রান হয়, তাহলে `globalThis` ব্যবহার করাই ভাল। 
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 5]
```diff
@@ -1,87 +1,87 @@
...
 ```
 
-In a browser, global functions and variables declared with `var` (not `let/const`!) become the property of the global object:
+ব্রাউজারে, গ্লোবাল ফাংশন এবং ভেরিয়েবল `var` দিয়ে (`let/const` নয়) ডিক্লেয়ার করা হয় এবং সেটি গ্লোবাল অবজেক্টের প্রপার্টি হয়ে যায়ঃ 
 
 ```js run untrusted refresh
 var gVar = 5;
 
-alert(window.gVar); // 5 (became a property of the global object)
+alert(window.gVar); // ৫ (গ্লোবাল অবজেক্টের প্রপার্টি হয়ে গিয়েছে )
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:25]:
```suggestion
alert(window.gVar); // ৫ (গ্লোবাল অবজেক্টের প্রপার্টি হয়ে গেছে)
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 6]
```diff
@@ -1,87 +1,87 @@
...
 ```
 
-Please don't rely on that! This behavior exists for compatibility reasons. Modern scripts use [JavaScript modules](info:modules) where such thing doesn't happen.
+দয়া করে এটির উপর নির্ভর করবেন না! এই বিহেবিয়ার কম্পাটিবিলিটির জন্য রাখা হয়েছে। আধুনিক স্ক্রিপ্টস [JavaScript modules](info:modules) ব্যবহার করে, যাতে এমনটা না হয়।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:28]:
```suggestion
দয়া করে এটির উপর নির্ভর করবেন না! এটি বিহেবিয়ার কম্পাটিবিলিটির জন্য রাখা হয়েছে। আধুনিক স্ক্রিপ্ট সমুহ [JavaScript modules](info:modules) ব্যবহার করে, যাতে এমনটা না হয়। 
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 7]
```diff
@@ -1,87 +1,87 @@
...
 
 ```js run untrusted refresh
 let gLet = 5;
 
-alert(window.gLet); // undefined (doesn't become a property of the global object)
+alert(window.gLet); // undefined (গ্লোবাল অবজেক্টের প্রপার্টি হয়ে যায়নি )
 ```
 
-If a value is so important that you'd like to make it available globally, write it directly as a property:
+যদি কোনও মান এত গুরুত্বপূর্ণ হয় যে আপনি এটি গ্লোবালি এভেইলাবল করতে চান, তাহলে তা সরাসরি প্রপার্টি হিসেবে লিখে দিতে পারেনঃ
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:38]:
```suggestion
যদি কোনও মান এত গুরুত্বপূর্ণ হয় যে আপনি এটি গ্লোবালি এভেইলেবল করতে চান, তাহলে তা সরাসরি প্রপার্টি হিসেবে লিখে দিতে পারেনঃ 
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 8]
```diff
@@ -1,87 +1,87 @@
...
 */!*
 
-// somewhere else in code
+// কোডের অন্য কোথাও
 alert(currentUser.name);  // John
 
-// or, if we have a local variable with the name "currentUser"
-// get it from window explicitly (safe!)
+// অথবা, আমাদের "currentUser" নামে যদি কোন লোকাল ভেরিয়েবল থাকে 
+// তাহলে সরাসরি window থেকে নেয়া যাবে (নিরাপদ!)
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:52]:
```suggestion
// তাহলে সরাসরি window থেকে নেয়া যাবে (এটা নিরাপদ!)
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 9]
```diff
@@ -1,87 +1,87 @@
...
 alert(window.currentUser.name); // John
 ```
 
-That said, using global variables is generally discouraged. There should be as few global variables as possible. The code design where a function gets "input" variables and produces certain "outcome" is clearer, less prone to errors and easier to test than if it uses outer or global variables.
+গ্লোবাল ভেরিয়েবলগুলি ব্যবহার করা সাধারণত নিরুৎসাহিত করা হয়। যতটা সম্ভব কম গ্লোবাল ভেরিয়েবল থাকা উচিত। যে কোড ডিজাইনে ফাংশন "ইনপুট" ভেরিয়েবল নেয় এবং কিছু নির্দিষ্ট আউটপুট তৈরি করে তা গ্লোবাল ভেরিয়েবলের তুলনায় পরিষ্কার, কম ত্রুটি প্রবণ এবং সহজে টেস্ট করা যায়।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:56]:
```suggestion
সাধারণত গ্লোবাল ভেরিয়েবলগুলি ব্যবহার করতে নিরুৎসাহিত করা হয়। যতটা সম্ভব কম গ্লোবাল ভেরিয়েবল থাকা উচিত। যে কোড ডিজাইনে ফাংশন "ইনপুট" এর মাধ্যমে ভেরিয়েবল নেয় এবং কিছু নির্দিষ্ট "আউটপুট" তৈরি করে তা গ্লোবাল ভেরিয়েবলের তুলনায় অধিক পরিষ্কার, কম ত্রুটি প্রবণ এবং সহজে টেস্ট করা যায়। 
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 10]
```diff
@@ -1,87 +1,87 @@
...
-For instance, test if a built-in `Promise` object exists (it doesn't in really old browsers):
+উদাহরণসরূপ, বিল্ডইন `Promise` অবজেক্ট আছে কিনা তা টেস্ট করা (পুরানো ব্রাউজারগুলিতে নেই):
 ```js run
 if (!window.Promise) {
   alert("Your browser is really old!");
 }
 ```
 
-If there's none (say, we're in an old browser), we can create "polyfills": add functions that are not supported by the environment, but exist in the modern standard.
+যদি না থাকে তাহলে আমরা "পলিফিলস্‌‌" তৈরি করতে পারিঃ ইনভার্মেন্টে নেই কিন্তু মর্ডান স্ট্যান্ডার্ডে আছে, এমন ফাংশন ইনভার্মেন্টে এড করা।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:69]:
```suggestion
যদি না থাকে তাহলে আমরা "পলিফিলস্‌‌" তৈরি করতে পারিঃ এনভাইরনমেন্টে নেই কিন্তু মর্ডান স্ট্যান্ডার্ডে আছে, এমন ফাংশন এনভাইরনমেন্টে সংযোজন করা। 
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 11]
```diff
@@ -1,87 +1,87 @@
...
-  window.Promise = ... // custom implementation of the modern language feature
+  window.Promise = ... // মর্ডান ল্যাঙ্গুইজ ফিচারের কাস্টম ইমপ্লিমেন্টেশন 
 }
 ```
 
-## Summary
+## সারাংশ
 
-- The global object holds variables that should be available everywhere.
+- গ্লোবাল অবজেক্টে যে ভেরিয়েবলগুলি থাকে তা সর্বত্র আভাইলাবল।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:79]:
```suggestion
- গ্লোবাল অবজেক্টে যে ভেরিয়েবলগুলি থাকে তা সর্বত্র আভেইলেবল।
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 12]
```diff
@@ -1,87 +1,87 @@
...
 
-    That includes JavaScript built-ins, such as `Array` and environment-specific values, such as `window.innerHeight` -- the window height in the browser.
-- The global object has a universal name `globalThis`.
+    এতে জাভাস্ক্রিপ্টের বিল্ডইন ফিচার অন্তর্ভুক্ত থাকে, যেমনঃ `Array` এবং ইনভার্মেন্ট এস্পেসিফিক ভ্যালু, `window.innerHeight` -- ব্রাউজারের উইন্ডো হাইট।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:81]:
```suggestion
    এতে জাভাস্ক্রিপ্টের বিলট-ইন ফিচারে অন্তর্ভুক্ত থাকে, যেমনঃ `Array` এবং এনভাইরনমেন্ট এস্পেসিফিক ভ্যালু, `window.innerHeight` -- ব্রাউজারের উইন্ডো হাইট। 
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 13]
```diff
@@ -1,87 +1,87 @@
...
-- The global object has a universal name `globalThis`.
+    এতে জাভাস্ক্রিপ্টের বিল্ডইন ফিচার অন্তর্ভুক্ত থাকে, যেমনঃ `Array` এবং ইনভার্মেন্ট এস্পেসিফিক ভ্যালু, `window.innerHeight` -- ব্রাউজারের উইন্ডো হাইট। 
+- গ্লোবাল অবজেক্টির সার্বজনীন নাম হল `globalThis`।
 
-    ...But more often is referred by "old-school" environment-specific names, such as `window` (browser) and `global` (Node.js). As `globalThis` is a recent proposal, it's not supported in non-Chromium Edge (but can be polyfilled).
-- We should store values in the global object only if they're truly global for our project. And keep their number at minimum.
-- In-browser, unless we're using [modules](info:modules), global functions and variables declared with `var` become a property of the global object.
-- To make our code future-proof and easier to understand, we should access properties of the global object directly, as `window.x`.
+    তবে এখনও বিভিন্ন ইনভার্মেন্টের পুরনো নামেই বেশি ডাকা হয়, যেমনঃ `window` (ব্রাউজার) এবং `global` (নোড জেএস)। `globalThis` হচ্ছে সাম্প্রতিক প্রস্তাব, এটি নন-ক্রোমিয়ামে সাপোর্ট করে না (তবে পলিফিল করা যাবে)।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:84]:
```suggestion
    তবে এখনও বিভিন্ন এনভাইরনমেন্টের পুরনো নামেই বেশি ডাকা হয়, যেমনঃ `window` (ব্রাউজার) এবং `global` (নোড জেএস)। `globalThis` হচ্ছে সাম্প্রতিক প্রস্তাব, এটি নন-ক্রোমিয়ামে সাপোর্ট করে না (তবে পলিফিল করা যাবে)।
```

File: 1-js/06-advanced-functions/05-global-object/article.md [hunk 14]
```diff
@@ -1,87 +1,87 @@
...
 
-    ...But more often is referred by "old-school" environment-specific names, such as `window` (browser) and `global` (Node.js). As `globalThis` is a recent proposal, it's not supported in non-Chromium Edge (but can be polyfilled).
-- We should store values in the global object only if they're truly global for our project. And keep their number at minimum.
-- In-browser, unless we're using [modules](info:modules), global functions and variables declared with `var` become a property of the global object.
-- To make our code future-proof and easier to understand, we should access properties of the global object directly, as `window.x`.
+    তবে এখনও বিভিন্ন ইনভার্মেন্টের পুরনো নামেই বেশি ডাকা হয়, যেমনঃ `window` (ব্রাউজার) এবং `global` (নোড জেএস)। `globalThis` হচ্ছে সাম্প্রতিক প্রস্তাব, এটি নন-ক্রোমিয়ামে সাপোর্ট করে না (তবে পলিফিল করা যাবে)। 
+- প্রজেক্টের কোন ভ্যালু যদি গ্লোবাল রাখার প্রয়োজন হয়, শুধুমাত্র সেই ভ্যালুগুলি গ্লোবাল অবজেক্টে ষ্টোর করতে হবে। এবং তার সংখ্যা কম রাখার চেষ্টা করতে হবে। 
+- ব্রাউজারে [modules](info:modules) ইউজ না করলে গ্লোবাল ফাংশন এবং ভ্যারিয়েবল `var` দিয়ে ডিক্লেয়ার করা হয়, যা গ্লোবাল অবজেক্টের প্রপার্টি হিসেবে এড হয়ে যায়।
```
Comment (jamal-pb95) [inline 1-js/06-advanced-functions/05-global-object/article.md:86]:
```suggestion
- ব্রাউজারে [modules](info:modules) ইউজ না করলে গ্লোবাল ফাংশন এবং ভ্যারিয়েবল `var` দিয়ে ডিক্লেয়ার করা হয়, যা গ্লোবাল অবজেক্টের প্রপার্টি হিসেবে সংযোজিত হয়ে যায়।
```

**Comment by raikusy** [comment]
@jamal-pb95 should we close this PR? as this has been so long with no updates.

**Comment by jamal-pb95** [comment]
hmm

**Comment by raikusy** [comment]
This PR is being closed, as there's no update from the author. Feel free to open a new PR if you get update on this. @AbmSourav

**Pull request closed without merging**